cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -2750,4 +2753,52 @@ public void setNotificationStyle(NotificationStyle style) {
         editor.putString(CleverPushPreferences.NOTIFICATION_STYLE, style.getCode());
         editor.apply();
     }
+
+    private void setUpNotificationCategoryGroups() {
+        getChannelConfig(new ChannelConfigListener() {
+            @Override
+            public void ready(JSONObject channelConfig) {
+                try {
+                    ArrayList<NotificationCategory> notificationCategories = new ArrayList<>();
+                    JSONArray notificationCategoryGroups = channelConfig.getJSONArray("notificationCategoryGroups");
+                    for (int i = 0; i < notificationCategoryGroups.length(); i++) {
+                        JSONObject notificationCategoryGroupJSONObject = notificationCategoryGroups.getJSONObject(i);
+                        NotificationCategoryGroup notificationCategoryGroup = new NotificationCategoryGroup();
+                        notificationCategoryGroup.setId(notificationCategoryGroupJSONObject.optString("_id"));
+                        notificationCategoryGroup.setName(notificationCategoryGroupJSONObject.optString("name"));
+                        JSONArray categories = notificationCategoryGroupJSONObject.getJSONArray("categories");
+                        for (int j = 0; j < categories.length(); j++) {
+                            JSONObject notificationCategoryJSONObject = categories.getJSONObject(j);
+
+                            NotificationCategory notificationCategory = new NotificationCategory();
+
+                            notificationCategory.setId(notificationCategoryJSONObject.optString("_id"));
+                            notificationCategory.setGroup(notificationCategoryGroup);
+                            notificationCategory.setName(notificationCategoryJSONObject.optString("name"));
```

Comment:
Can't we use `Gson` here instead of getting all the values from JSONObject manually?

Reply:
Otherwise please also put it into a util func so it does not make the main class longer

Reply:
We can't use Gson as it will expect id and name for category group but in response we have string instead of json object 
will move it to util